immich-app/immich
# feat(server): delete unnecessary encoded videos

## Description

The current transcoding behavior only allows for the creation or replacement of encoded videos, but not their deletion. This means if an admin decides to change their transcode policy after videos have been transcoded, previous transcodes will continue to exist even if the new policy doesn't require them. This PR adds a check to the transcoding job handler to delete encoded videos in cases where there is an existing transcode despite the current policy.

Fixes #5873


## How Has This Been Tested?

1. Inspect the `encoded-video` directory with `find encoded-video -name *.mp4`, confirming there are videos there
2. Set the transcode policy to disabled
3. Run transcoding jobs for all videos
4. Confirm that the microservices container has logs about deleting transcodes
5. Confirm that  `find encoded-video -name *.mp4`  does not return any videos
